packages/amplify-provider-awscloudformation/src/iterative-deployment/helpers.ts

```diff
@@ -0,0 +1,50 @@
+import { DeployMachineContext, DeploymentMachineOp } from './state-machine';
+import * as path from 'path';
+
+export const hasMoreRollback = (context: DeployMachineContext) => {
+  return context.currentIndex >= 0;
+};
+
+export const hasMoreDeployment = (context: DeployMachineContext) => {
+  return context.stacks.length > context.currentIndex;
+};
+
+export const stackPollerActivity = (
+  stackEventPollFn: (stack: Readonly<DeploymentMachineOp>) => () => void,
+  operation: 'deploying' | 'rollingback',
+) => {
+  return (context: Readonly<DeployMachineContext>) => {
+    if (context.currentIndex >= 0 && context.currentIndex < context.stacks.length) {
+      const stack = context.stacks[context.currentIndex];
+      const step = operation == 'deploying' ? stack.deployment : stack.rollback;
+
+      return stackEventPollFn(step);
+    }
+    return () => {};
+  };
+};
+
+export const extractStackInfoFromContext = (
```

Comment:
I think a better name here would be `getOperationHandler`

I'm also not a huge fan of passing in the deploying/rollingback flag into this function. Perhaps it can be aliased like
```
export getDeploymentOperationHandler = fn => getOperationHandler(fn, 'deploying');
export getRollbackOperationHandler = fn => getOperationHandler(fn, 'rollingback');
```
and then don't export getOperationHandler so that callers can just call the method they want rather than passing in the flag